frontend/packages/topology/src/components/quick-search/QuickSearchController.tsx

```diff
@@ -1,23 +1,68 @@
 import * as React from 'react';
-import { CatalogService } from '@console/dev-console/src/components/catalog/service/CatalogServiceProvider';
+import { useTranslation } from 'react-i18next';
 import { getQueryArgument } from '@console/internal/components/utils';
+import { quickSearch } from './utils/quick-search-utils';
 import QuickSearchButton from './QuickSearchButton';
 import QuickSearchModal from './QuickSearchModal';
+import { QuickSearchData, QuickSearchProviders } from './utils/quick-search-types';
+import { Button } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
 
-type QuickSearchControllerProps = CatalogService & {
+type QuickSearchControllerProps = {
   namespace: string;
   viewContainer?: HTMLElement;
+  quickSearchProviders: QuickSearchProviders;
+  allItemsLoaded: boolean;
 };
 
 const QuickSearchController: React.FC<QuickSearchControllerProps> = ({
   namespace,
-  searchCatalog,
-  loaded,
+  quickSearchProviders,
   viewContainer,
+  allItemsLoaded,
 }) => {
   const [isQuickSearchActive, setIsQuickSearchActive] = React.useState<boolean>(
     !!getQueryArgument('catalogSearch'),
   );
+  const { t } = useTranslation();
+
+  const searchCatalog = React.useCallback(
+    (searchTerm: string): QuickSearchData => {
+      return quickSearchProviders.reduce(
+        (acc, quickSearchProvider) => {
+          const items = quickSearchProvider.loaded
+            ? quickSearch(quickSearchProvider.items, searchTerm)
+            : [];
+          const itemCount = items.length;
+          const viewAllLink =
+            itemCount > 0
+              ? [
+                  <Button
+                    variant="link"
+                    isInline
+                    tabIndex={-1}
+                    key={quickSearchProvider.catalogType}
+                  >
+                    <Link to={quickSearchProvider.getCatalogURL(searchTerm, namespace)}>
```

Comment:
This is very odd to wrap the `Link` with a `Button`. I guess PF doesn't have good support for this. 
Ideally use `<Button component={Link} to={...}` however PF doesn't allow us to provide the proper types to the Button.

I'd prefer to work around the typescript types incompatibility while we sort out a fix with PF instead of creating the odd markup.

Reply:
Raised a PF issue: https://github.com/patternfly/patternfly-react/issues/5491

Reply:
Tried the typescript workaround with sending link in component prop of button component. It worked with some style issues but the tab enter didn’t work correctly. Now using <Link> directly instead of wrapping it inside <Button>